zio-crypto/shared/src/main/scala/zio/crypto/symmetric/SymmetricEncryption.scala

```diff
@@ -0,0 +1,120 @@
+package zio.crypto.symmetric
+
+import zio._
+import zio.crypto.random.SecureRandom
+import zio.crypto.random.SecureRandom.SecureRandom
+
+import javax.crypto.spec.GCMParameterSpec
+import javax.crypto.{ Cipher, KeyGenerator, SecretKey }
+
+sealed trait SymmetricEncryptionAlgorithm
+
+object SymmetricEncryptionAlgorithm {
+  case object AES128 extends SymmetricEncryptionAlgorithm
+  case object AES192 extends SymmetricEncryptionAlgorithm
+  case object AES256 extends SymmetricEncryptionAlgorithm
+}
+
+case class CipherText(value: Array[Byte])           extends AnyVal
+case class SymmetricEncryptionKey(value: SecretKey) extends AnyVal
+
+object SymmetricEncryption {
+
+  type SymmetricEncryption = Has[SymmetricEncryption.Service]
+
+  trait Service {
+    def encrypt(plainText: Array[Byte], key: SymmetricEncryptionKey): RIO[SecureRandom, CipherText]
+    def decrypt(ciphertext: CipherText, key: SymmetricEncryptionKey): Task[Array[Byte]]
+    def getKey(alg: SymmetricEncryptionAlgorithm): RIO[SecureRandom, SymmetricEncryptionKey]
+  }
+
+  val live: ULayer[SymmetricEncryption] = ZLayer.succeed(new Service {
+
+    /**
+     * In our implementation, we will use the most secure tag size as defined
+     * by: http://nvlpubs.nist.gov/nistpubs/Legacy/SP/nistspecialpublication800-38d.pdf
+     * Iv length of 96 bits is recommended as per the spec on page 8
+     */
+    val NISTTagLengthBits = 128
+    val NISTIvLengthBytes = 12
+
+    private def getInstance: Task[Cipher] =
+      Task.effect(Cipher.getInstance("AES/GCM/NoPadding"))
```

Comment:
This will not be cached. Every time `getInstance` is sequenced into a computation, it will be called again. Does this have potential to be a performance problem?

Reply:
My understanding is that `Cipher` is a mutable object that you use for a single signature, and that using it more than once would result in an error.